torch/csrc/autograd/python_engine.cpp

```diff
@@ -120,14 +124,14 @@ PyObject *THPEngine_run_backward(THPEngine *self, PyObject *args, PyObject *kwar
     // see "suppress grad accumulation" below. Note that only variables which
     // have requires_grad=True can have grad accumulators.
     auto grad_fn = variable.grad_fn() ? variable.grad_fn() : variable.grad_accumulator();
```

Comment:
This code is very similar to `FunctionPort::get_gradient` but without some define/requires_grad checks. Maybe make an unchecked version of `FunctionPort.get_gradient` to encapsulate these cases.